src/crypto/secp156k1_provider/secp256k1_provider_impl.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "libp2p/crypto/secp256k1_provider/secp256k1_provider_impl.hpp"
+
+#include <memory>
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/obj_mac.h>
+#include "libp2p/crypto/error.hpp"
+#include "libp2p/crypto/sha/sha256.hpp"
+
+namespace libp2p::crypto::secp256k1 {
+  outcome::result<KeyPair> Secp256k1ProviderImpl::generateKeyPair() const {
+    PublicKey publicKey{};
+    PrivateKey privateKey{};
+    std::shared_ptr<EC_KEY> key{EC_KEY_new_by_curve_name(NID_secp256k1),
+                                EC_KEY_free};
+    if (EC_KEY_generate_key(key.get()) != 1) {
+      return KeyGeneratorError::KEY_GENERATION_FAILED;
+    }
+    const BIGNUM *privateNum = EC_KEY_get0_private_key(key.get());
+    if (BN_bn2binpad(privateNum, privateKey.data(), privateKey.size()) < 0) {
+      return KeyGeneratorError::KEY_GENERATION_FAILED;
+    }
+    EC_KEY_set_conv_form(key.get(), POINT_CONVERSION_COMPRESSED);
+    auto publicKeyLength = i2o_ECPublicKey(key.get(), nullptr);
+    if (publicKeyLength != publicKey.size()) {
+      return KeyGeneratorError::KEY_GENERATION_FAILED;
+    }
+    uint8_t *publicKeyPtr = publicKey.data();
```

Comment:
check with clang-tidy, it may fail

Reply:
Clang-tidy is performed during CI, all checks are successful